PubHub/PubHub.API/Controllers/GenresController.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Net.Mime;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using PubHub.API.Controllers.Problems;
+using PubHub.API.Domain;
+using PubHub.API.Domain.Entities;
+using PubHub.Common.Models.Genres;
+using static PubHub.Common.IntegrityConstants;
+
+namespace PubHub.API.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError, Type = typeof(ProblemDetails))]
+    public class GenresController : ControllerBase
+    {
+        private readonly PubHubContext _context;
+
+        public GenresController(PubHubContext context)
+        {
+            _context = context;
+        }
+
+        [HttpGet()]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IList<GenreInfoModel>))]
+        public async Task<IResult> GetGenresAsync()
+        {
+            var genres = await _context.Set<Genre>()
+                .Select(genre => new GenreInfoModel
+                {
+                    Id = genre.Id,
+                    Name = genre.Name
+                })
+                .ToListAsync();
+
+            return Results.Ok(genres);
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(GenreInfoModel))]
+        [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(ProblemDetails))]
+        public async Task<IResult> GetGenresAsync(Guid id)
```

Comment:
Shouldn't this be called `GetGenreAsync()` ?

Reply:
I can see it's the same in the `AuthorController`, so if we want to have it singular, I can change that one when I make the `AuthorService`

Reply:
You are correct. I'll fix that when I get home 👌

Reply:
Alrighty. I will fix it in the `AuthorController` on my `AuthorService` branch then - unless you will on this? 😊

Reply:
Should be fixed now, @jeongoks